### PR TITLE
docs: fix spelling mistakes in many docs files

### PR DESCRIPTION
**Pull request by jcantosz**
Noticed some spelling mistakes while reading the argo docs, decided to go through and find and fix as many of them as I could so I don't have open up PRs while reading 😄 
Signed-off-by: Jesse Antoszyk <22500761+jcantosz@users.noreply.github.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] ~I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.~ N/A
* [x] ~I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.~ N/A
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] ~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~ N/A
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
You're an absolute legend. Would you kindly run `make codegen`? (Maybe start with `make clidocsgen` - it'll be faster, and I think that will cover everything.)

**Review by crenshaw-dev — APPROVED** [review]
Love it. Will merge after codegen. Thanks!

File: docs/operator-manual/notifications/services/rocketchat.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ stringData:
   rocketchat-password: <password>
 ```
 
-8. Finally, use these credentials to configure the RocketChat integration in the `argocd-configmap` config map:
+8. Finally, use these credentials to configure the RocketChat integration in the `argocd-configmap` config map:
```
Comment (crenshaw-dev) [inline docs/operator-manual/notifications/services/rocketchat.md:40]: Weird. Not sure why `make codegen` would cause a regression.

Comment (jcantosz) [reply docs/operator-manual/notifications/services/rocketchat.md:40]: Looks like the `codegen` reverted a few those changes. It looks like these are generated from a [dependency](https://github.com/argoproj/argo-cd/blob/master/cmd/argocd/commands/admin/notifications.go) -- I will create a a pr for the notifications-engine

**Comment by jcantosz** [comment]
Awesome! Thank you for the `make codegen` pointer. I think we are past the codegen error

I have used these projects a ton, seems the least I can do 😄

**Pull request merged**